autopilot/contractor.go

```diff
@@ -253,21 +254,25 @@ func (c *contractor) runContractChecks(cfg Config, contracts []worker.Contract)
 		}
 
 		// decide whether the contract is still good
-		usable, renewable, reasons := isUsableContract(cfg, Host{host}, contract, c.blockHeight)
+		usable, refresh, renew, reasons := isUsableContract(cfg, Host{host}, contract, c.blockHeight)
 		if !usable {
 			c.logger.Infow(
 				"unusable contract",
 				"hk", host.PublicKey,
 				"fcid", contract.ID,
 				"reasons", reasons,
-				"renewable", renewable,
+				"refresh", refresh,
+				"renew", renew,
 			)
-			if !renewable {
-				toDelete = append(toDelete, contract.ID)
-				continue
-			} else {
+			if refresh {
+				refreshIndices[contract.ID] = len(toRefresh)
+				toRefresh = append(toRefresh, contract)
+			} else if renew {
```

Comment:
I think this might be the wrong prioritisation. If a contract is `renew == true` we are in the renew window. So we prefer to renew it over refreshing it because otherwise we are going to refresh it and 10 minutes later renew it.